src/libraries/System.Private.CoreLib/src/System/IO/FileStatus.OSX.cs

```diff
@@ -0,0 +1,67 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Runtime.InteropServices;
+
+namespace System.IO
+{
+    internal partial struct FileStatus
+    {
+        internal unsafe void SetCreationTime(string path, DateTimeOffset time)
```

Comment:
Can you rename this file to `FileStatus.[Creation]Time.OSX.cs` and add a file with the non-OSX implementation of these methods named `FileStatus.[Creation]Time.Unix.cs`.

Reply:
See https://github.com/dotnet/runtime/pull/49555#discussion_r705040917. Area owners said they'd like to avoid code duplication which is why there is `.OSX`, `.OtherUnix` and `.Unix`.

Reply:
I was under that impression anyway. It's because it would require maintaining 2 lots of code that are the same, but if they would like me to move it I can. I'd probably name the files `FileStatus.SetTimes.OSX.cs` and `FileStatus.SetTimes.OtherUnix.cs` though as it was said somewhere that it should be `OtherUnix` rather than `Unix` I think.

Reply:
> I was under that impression anyway. It's because it would require maintaining 2 lots of code that are the same, but if they would like me to move it I can.

Adding the `.SetTimes` (or similar) in the name indicates these files are specific to that function.

There shouldn't be code duplication, the common code should be in `FileStatus.Unix.cs`.

> as it was said somewhere that it should be OtherUnix rather than Unix I think.

There are no `OtherUnix` files in the repo so far, that's why I suggested `Unix`. `OtherUnix` is also fine for me.

Reply:
The code duplication comment was related to https://github.com/dotnet/runtime/pull/49555#discussion_r705032835. I'll rename the files to `FileStatus.SetTimes.OSX.cs` and `FileStatus.SetTimes.OtherUnix.cs`. Source of `OtherUnix` is https://github.com/dotnet/runtime/issues/39132#issuecomment-657303253.